packages/patternfly-4/react-core/src/components/LoginPage/LoginPage.tsx

```diff
@@ -1,99 +1,65 @@
-import React from 'react';
+import * as React from 'react';
 import { css } from '@patternfly/react-styles';
-import PropTypes from 'prop-types';
 
-import Login from './Login';
-import LoginHeader from './LoginHeader';
-import LoginFooter from './LoginFooter';
-
-import LoginMainHeader from './LoginMainHeader';
-import LoginMainBody from './LoginMainBody';
-import LoginMainFooter from './LoginMainFooter';
-
-import { BackgroundImage } from '../BackgroundImage';
+import { BackgroundImage, BackgroundImageSrcMap } from '../BackgroundImage';
 import { Brand } from '../Brand';
 import { List } from '../List';
 
-const LoginListVariant = {
-  grid: 'grid',
-  inline: 'inline'
-};
+import { Login } from './Login';
+import { LoginHeader } from './LoginHeader';
+import { LoginFooter } from './LoginFooter';
+import { LoginMainHeader } from './LoginMainHeader';
+import { LoginMainBody } from './LoginMainBody';
+import { LoginMainFooter } from './LoginMainFooter';
 
-const propTypes = {
+export interface LoginPageProps extends React.HTMLProps<HTMLDivElement> {
   /** Anything that can be rendered inside of the LoginPage (e.g. <LoginPageForm>) */
-  children: PropTypes.node,
+  children?: React.ReactNode;
   /** Additional classes added to the LoginPage. */
-  className: PropTypes.string,
+  className?: string;
   /** Attribute that specifies the URL of the brand image for the LoginPage */
-  brandImgSrc: PropTypes.string,
+  brandImgSrc?: string;
   /** Attribute that specifies the alt text of the brand image for the LoginPage. */
-  brandImgAlt: PropTypes.string,
+  brandImgAlt?: string;
   /** Attribute that specifies the URL of the background image for the LoginPage */
-  backgroundImgSrc: PropTypes.oneOfType([
-    PropTypes.string,
-    PropTypes.shape({
-      xs: PropTypes.string,
-      xs2x: PropTypes.string,
-      sm: PropTypes.string,
-      sm2x: PropTypes.string,
-      lg: PropTypes.string,
-      filter: PropTypes.string
-    })
-  ]),
+  backgroundImgSrc?: string | BackgroundImageSrcMap;
```

Comment:
I am not sure this will work for documentation due to the ` string` in the union.  We should update this to use the string values for BackgroundImageSrcMap.

Reply:
@tlabaj My assumption, based on the code in BackgroundImage, is that there are potentially other strings that might be valid as wel (otherwise, why have the `string | `?). I can change the `BackgroundImageSrcMap` part to list the options, but with any string being valid it will still show as `union` in the docs, correct?

Reply:
Ye, that is my understanding as well.

Reply:
so is it worth listing the values here? If it's going to say `union` anyhow, might as well use the enum here (I assume we would be using the enums everywhere if the docs worked correctly).